### ISSUE TITLE
Issue with gender being 1.5 in individual file

### ISSUE THREAD

**Issue by Moohan** [Public-Health-Scotland/source-linkage-files#276]
From @Ryan-Harris93

>Just a quick point on the SLFs as Gaby was working with the files earlier this morning.

>When Gaby was completing the translation for the population & pathways matrix earlier today she spotted an issue with the gender classification in the older source individual files. I have pasted in a frequency example below for 2016/17
 
![image](https://user-images.githubusercontent.com/5982260/175492807-3d8e4e38-565e-4996-a00c-c8641eebdbfa.png)

>From what Gaby explained to me earlier, she mentioned that there were some gender codes listed as 1.5, which SPSS rounds to 2, shown in the frequency table above, but not defined as male or female as the actual value isn’t 1 or 2, hence the separate frequency. However in R, these numbers are not rounded, so they remained as 1.5 during the frequency check, which is where she spotted the difference when comparing against each other after an aggregate.

>We know this is the case for 2016/17 and 2015/16 so maybe worth checking for years prior to this, granted it’s a small number but I assume it’s still worthwhile changing. Everything checks out fine from 2017/18 onwards.

**Comment by Moohan** [comment]
The issue is in the D01 syntax. We take the 'most frequently appearing' gender, but if there isn't one i.e. the average comes out at `1.5` then we try to take it from the CHI number.

The problem (I assume) is when that person doesn't have a CHI number.

**Comment by Jennit07** [comment]
I'm also not too sure what is causing the issue here. I have tested 2016/17 in many ways with frequencies before/after running the code. `rnd(gender)` appears to be working fine with returning either 1 or 2 before adding male and female labels. I thought there may be something further down the line that caused this so also tested running this to the end of D05. I checked frequencies after each 'Temp' file and this always returned 1 and 2 values. This may be the case of running the individual files again if this only affects the older files?
